client/src/main/java/org/apache/cassandra/sidecar/client/RequestContext.java

```diff
@@ -0,0 +1,455 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.client;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.cassandra.sidecar.client.request.CleanSSTableUploadSessionRequest;
+import org.apache.cassandra.sidecar.client.request.ClearSnapshotRequest;
+import org.apache.cassandra.sidecar.client.request.CreateSnapshotRequest;
+import org.apache.cassandra.sidecar.client.request.GossipInfoRequest;
+import org.apache.cassandra.sidecar.client.request.ImportSSTableRequest;
+import org.apache.cassandra.sidecar.client.request.ListSnapshotFilesRequest;
+import org.apache.cassandra.sidecar.client.request.NodeSettingsRequest;
+import org.apache.cassandra.sidecar.client.request.Request;
+import org.apache.cassandra.sidecar.client.request.RingRequest;
+import org.apache.cassandra.sidecar.client.request.SSTableComponentRequest;
+import org.apache.cassandra.sidecar.client.request.SchemaRequest;
+import org.apache.cassandra.sidecar.client.request.TimeSkewRequest;
+import org.apache.cassandra.sidecar.client.request.TokenRangeReplicasRequest;
+import org.apache.cassandra.sidecar.client.request.UploadSSTableRequest;
+import org.apache.cassandra.sidecar.client.retry.ExponentialBackoffRetryPolicy;
+import org.apache.cassandra.sidecar.client.retry.NoRetryPolicy;
+import org.apache.cassandra.sidecar.client.retry.RetryPolicy;
+import org.apache.cassandra.sidecar.client.selection.InstanceSelectionPolicy;
+import org.apache.cassandra.sidecar.client.selection.SingleInstanceSelectionPolicy;
+import org.apache.cassandra.sidecar.common.utils.HttpRange;
+
+/**
+ * The context for a given request that include the {@link InstanceSelectionPolicy}, the {@link RetryPolicy}, and
+ * the {@link Request}
+ */
+public class RequestContext
+{
+    protected static final SchemaRequest FULL_SCHEMA_REQUEST = new SchemaRequest();
+    protected static final TimeSkewRequest TIME_SKEW_REQUEST = new TimeSkewRequest();
+    protected static final NodeSettingsRequest NODE_SETTINGS_REQUEST = new NodeSettingsRequest();
+    protected static final RingRequest RING_REQUEST = new RingRequest();
```

Comment:
nit: the request objects, as well as the request builder methods below, do not feel should part of `RequestContext`, which supplies the context for a request. 
But I think it is fine to leave them as is for now.

Reply:
maybe we can have it as a follow up